src/core/permissions/Permission.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+
+import { Enum } from 'enumify';
+
+function enumFactory(friendlyName) {
+  return {
+    get friendlyName() {
+      return friendlyName;
+    },
+    hasPermission(permission: Permission) {
+      return this.ordinal >= permission.ordinal;
+    }
+  };
+}
+
+export class Permission extends Enum {}
+Permission.initEnum({
+  'HIDDEN': enumFactory("Hidden"),
+  'DISCOVER': enumFactory("Discover"),
+  'READ': enumFactory("Read"),
+  'WRITE': enumFactory("Write"),
+  'OWNER': enumFactory("Owner")
+});
+
+export function maxPermission(permissions: Permission) {
+  let highestPerm = Permission.HIDDEN;
+
+  for (let permission of permissions) {
+    if (permission.ordinal > highestPerm.ordinal) {
+      highestPerm = permission;
```

Comment:
I did :p. This function is anticipating how permissions are going to be managed on the front-end in the future. Arguably, I shouldn't have added it until it was used, but we'll use it in a future PR